src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Buffers
+{
+    internal sealed class DefaultArrayPool<T> : ArrayPool<T>
+    {
+        private const int MinimiumArraySize = 16;
+        private int _maxArrayLength;
+        private int _numberOfArraysPerBucket;
```

Comment:
There are various potential optimizations the backend compiler can employ with readonly.  For example, with a readonly static of a primitive type, the JIT may actually be able to effectively turn it into a constant and fold its result into call sites it later compiles.  For an instance field like this, I don't know if there are any optimizations applied by the JIT, but if nothing else it helps with code understanding, maintainability, etc.